flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManager.java

```diff
@@ -86,7 +88,7 @@ public interface SlotManager extends AutoCloseable {
     void start(
             ResourceManagerId newResourceManagerId,
             Executor newMainThreadExecutor,
-            ResourceActions newResourceActions,
+            @Nullable ResourceActions newResourceActions,
```

Comment:
I would not make this argument `@Nullable`.

I'm aware that many fields, including `resourceActions`, are `@Nullable` in the slot manager implementations. However, that is because these fields can only be decided when the slot manager is started, not when constructed. The null values can be ambiguous. E.g., in this case, `resourceActions` being `null` can mean either the slot manager is not yet started, or the resource allocation is not supported.

As an alternative, I'd suggest to introduce another implementation of `ResourceActions`, something like a `NonSupportedResourceActions`. And we can add a new method `ResourceActions#isSupported`, and use it for deciding whether the resource can be allocated. For the `NonSupportedResourceActions`, `allocateResource` and `releaseResource` should throw `UnsupportedOperationException`, meaning that the slot manager is responsible for calling these two methods only on when `ResourceActions#isSupported` returns `true`.